src/lib/Makefile

```diff
@@ -5,69 +5,53 @@
-PARCSRLSFILES = ${HYPRE_SRC_TOP_DIR}/parcsr_ls/*.o
-PARCSRMVFILES = ${HYPRE_SRC_TOP_DIR}/parcsr_mv/*.o
+FEIHYPREFILES      = ${HYPRE_FEI_HYPRE_FILES}
+FEMLIFILES         = ${HYPRE_FEI_FEMLI_FILES}
+IJMVFILES          = ${HYPRE_SRC_TOP_DIR}/IJ_mv/*.o ${HYPRE_SRC_TOP_DIR}/IJ_mv/*.obj
+EUCLIDFILES        = ${HYPRE_EUCLID_FILES}
+PARASAILSFILES     = ${HYPRE_PARASAILS_FILES}
+PILUTFILES         = ${HYPRE_PILUT_FILES}
+DISTMATRIXFILES    = ${HYPRE_SRC_TOP_DIR}/distributed_matrix/*.o
+KRYLOVFILES        = ${HYPRE_SRC_TOP_DIR}/krylov/*.o
+MATMATFILES        = ${HYPRE_SRC_TOP_DIR}/matrix_matrix/*.o
+MULTIVECFILES      = ${HYPRE_SRC_TOP_DIR}/multivector/*.o
+PARCSRLSFILES      = ${HYPRE_SRC_TOP_DIR}/parcsr_ls/*.o ${HYPRE_SRC_TOP_DIR}/parcsr_ls/*.obj
+PARCSRMVFILES      = ${HYPRE_SRC_TOP_DIR}/parcsr_mv/*.o ${HYPRE_SRC_TOP_DIR}/parcsr_mv/*.obj
 PARCSRBLOCKMVFILES = ${HYPRE_SRC_TOP_DIR}/parcsr_block_mv/*.o
-SEQMVFILES = ${HYPRE_SRC_TOP_DIR}/seq_mv/*.o
-SSTRUCTLSFILES = ${HYPRE_SRC_TOP_DIR}/sstruct_ls/*.o
-SSTRUCTMVFILES = ${HYPRE_SRC_TOP_DIR}/sstruct_mv/*.o
-STRUCTLSFILES = ${HYPRE_SRC_TOP_DIR}/struct_ls/*.o
-STRUCTMVFILES = ${HYPRE_SRC_TOP_DIR}/struct_mv/*.o
-UTILITIESFILES = ${HYPRE_SRC_TOP_DIR}/utilities/*.o
-BLASFILES = ${HYPRE_SRC_TOP_DIR}/blas/*.o
-LAPACKFILES = ${HYPRE_SRC_TOP_DIR}/lapack/*.o
-
-IJMVOBJFILES = ${HYPRE_SRC_TOP_DIR}/IJ_mv/*.obj
-PARCSRLSOBJFILES = ${HYPRE_SRC_TOP_DIR}/parcsr_ls/*.obj
-PARCSRMVOBJFILES = ${HYPRE_SRC_TOP_DIR}/parcsr_mv/*.obj
-SEQMVOBJFILES = ${HYPRE_SRC_TOP_DIR}/seq_mv/*.obj
-SSTRUCTLSOBJFILES = ${HYPRE_SRC_TOP_DIR}/sstruct_ls/*.obj
-SSTRUCTMVOBJFILES = ${HYPRE_SRC_TOP_DIR}/sstruct_mv/*.obj
-STRUCTLSOBJFILES = ${HYPRE_SRC_TOP_DIR}/struct_ls/*.obj
-STRUCTMVOBJFILES = ${HYPRE_SRC_TOP_DIR}/struct_mv/*.obj
-UTILITIESOBJFILES = ${HYPRE_SRC_TOP_DIR}/utilities/*.obj
+SEQMVFILES         = ${HYPRE_SRC_TOP_DIR}/seq_mv/*.o ${HYPRE_SRC_TOP_DIR}/seq_mv/*.obj
+SEQBLOCKMVFILES    = ${HYPRE_SRC_TOP_DIR}/seq_block_mv/*.o
+SSTRUCTLSFILES     = ${HYPRE_SRC_TOP_DIR}/sstruct_ls/*.o ${HYPRE_SRC_TOP_DIR}/sstruct_ls/*.obj
+SSTRUCTMVFILES     = ${HYPRE_SRC_TOP_DIR}/sstruct_mv/*.o ${HYPRE_SRC_TOP_DIR}/sstruct_mv/*.obj
+STRUCTLSFILES      = ${HYPRE_SRC_TOP_DIR}/struct_ls/*.o ${HYPRE_SRC_TOP_DIR}/struct_ls/*.obj
+STRUCTMVFILES      = ${HYPRE_SRC_TOP_DIR}/struct_mv/*.o ${HYPRE_SRC_TOP_DIR}/struct_mv/*.obj
+UTILITIESFILES     = ${HYPRE_SRC_TOP_DIR}/utilities/*.o ${HYPRE_SRC_TOP_DIR}/utilities/*.obj
+BLASFILES          = ${HYPRE_SRC_TOP_DIR}/blas/*.o
+LAPACKFILES        = ${HYPRE_SRC_TOP_DIR}/lapack/*.o
 
 FILES_HYPRE = \
-$(FEIHYPREFILES)\
-$(FEMLIFILES)\
-$(IJMVFILES)\
-$(EUCLIDFILES)\
-$(PARASAILSFILES)\
-$(PILUTFILES)\
-$(DISTMATRIXFILES)\
-$(KRYLOVFILES)\
-$(MATMATFILES)\
-$(MULTIVECFILES)\
-$(PARCSRLSFILES)\
-$(PARCSRMVFILES)\
-$(PARCSRBLOCKMVFILES)\
-$(SEQMVFILES)\
-$(SSTRUCTLSFILES)\
-$(SSTRUCTMVFILES)\
-$(STRUCTLSFILES)\
-$(STRUCTMVFILES)\
-$(UTILITIESFILES)\
-$(BLASFILES)\
-$(LAPACKFILES)\
-$(IJMVOBJFILES)\
```

Comment:
Are you sure you can delete these?  We used to have to do something special for the '.obj' files so I think these were needed at one time.

Reply:
Yes, this is fine. I moved the `.obj` files to their respective folders, e.g., the `.obj` files under `$(IJMVOBJFILES)` are now in `$(IJMVFILES)`. This allowed to reduce some lines of code... 

GPU compilation after this change looks fine (tested on lassen and tioga)

Reply:
I see.  You might still want to check with Rui Peng just in case he had another reason for organizing things the way he did.  I remember he and I chatting about this at the time, but I don't recall the details.

Reply:
@liruipeng can you verify this particular change looks good?

Reply:
I guess it's OK assuming it's tested. I don't see or remember there has to be a particular order for the .o/.objs.